ibis/expr/types/generic.py

```diff
@@ -119,7 +140,479 @@ def _repr_html_(self) -> str | None:
+        SimpleCase[string*]
+          base:
+            string_col = Column[string*] 'string_col' from table
+              ref_0
+          cases:
+            Literal[string]
+              a
+            Literal[string]
+              b
+          results:
+            Literal[string]
+              an a
+            Literal[string]
+              a b
+          default:
+            Literal[string]
+              null or (not a and not b)
+        """
+        import ibis.expr.builders as bl
+
+        return bl.SimpleCaseBuilder(self)
+
+    def cases(
+        self,
+        case_result_pairs: Iterable[tuple[ir.BooleanValue, ValueExpr]],
+        default: ValueExpr | None = None,
+    ) -> ValueExpr:
+        """Create a case expression in one shot.
+
+        Parameters
+        ----------
+        case_result_pairs
+            Conditional-result pairs
+        default
+            Value to return if none of the case conditions are true
+
+        Returns
+        -------
+        ValueExpr
+            Value expression
+        """
+        builder = self.case()
+        for case, result in case_result_pairs:
+            builder = builder.when(case, result)
+        return builder.else_(default).end()
+
+    def collect(self) -> ir.ArrayValue:
+        """Return an array of the elements of this expression."""
+        import ibis.expr.operations as ops
+
+        return ops.ArrayCollect(self).to_expr()
+
+    def identical_to(self, other: ValueExpr) -> ir.BooleanValue:
+        """Return whether this expression is identical to other.
+
+        Corresponds to `IS NOT DISTINCT FROM` in SQL.
+
+        Parameters
+        ----------
+        other
+            Expression to compare to
+
+        Returns
+        -------
+        BooleanValue
+            Whether this expression is not distinct from `other`
+        """
+        import ibis.expr.operations as ops
+        import ibis.expr.rules as rlz
+
+        try:
+            return ops.IdenticalTo(self, rlz.any(other)).to_expr()
+        except (com.IbisTypeError, NotImplementedError):
+            return NotImplemented
+
+    def group_concat(
+        self,
+        sep: str = ",",
+        where: ir.BooleanValue | None = None,
+    ) -> ir.StringScalar:
+        """Concatenate values using the indicated separator to produce a string.
+
+        Parameters
+        ----------
+        sep
+            Separator will be used to join strings
+        where
+            Filter expression
+
+        Returns
+        -------
+        StringScalar
+            Concatenated string expression
+        """
+        import ibis.expr.operations as ops
+
+        return ops.GroupConcat(self, sep=sep, where=where).to_expr()
+
+    def __hash__(self) -> int:
+        return super().__hash__()
+
+    def __eq__(self, other: AnyValue) -> ir.BooleanValue:
+        import ibis.expr.operations as ops
+        import ibis.expr.rules as rlz
+
+        return _binop(ops.Equals, self, rlz.any(other))
```

Comment:
Why is `rlz.any()` required here?

Reply:
It may not be, I think was copying the existing code.

Reply:
This is necessary for `_maybe_cast_args`. Changing that is out of scope for this PR.